src/main/java/io/fabric8/maven/docker/service/BuildService.java

```diff
@@ -60,14 +61,55 @@
      * @throws DockerAccessException
      * @throws MojoExecutionException
      */
-    public void buildImage(ImageConfiguration imageConfig, ImagePullManager imagePullManager, BuildContext buildContext)
+    public void buildImage(ImageConfiguration imageConfig, ImagePullManager imagePullManager, BuildContext buildContext, File buildArchiveFile)
             throws DockerAccessException, MojoExecutionException {
 
         if (imagePullManager != null) {
             autoPullBaseImage(imageConfig, imagePullManager, buildContext);
         }
 
-        buildImage(imageConfig, buildContext.getMojoParameters(), checkForNocache(imageConfig), addBuildArgs(buildContext));
+        buildImage(imageConfig, buildContext.getMojoParameters(), checkForNocache(imageConfig), addBuildArgs(buildContext), buildArchiveFile);
+    }
+
+    /**
+     * Create docker archive for building image
+     *
+     * @param imageConfiguration image configuration
+     * @param buildContext docker build context
+     * @param buildArchiveOnlyFlagOrBuildArchiveLocation build archive only flag, it can have values TRUE or FALSE and also
+     *                               it can hold path to archive where it might get copied over
+     * @return tarball for docker image
+     * @throws MojoExecutionException in case any exception comes during building tarball
+     */
+    public File buildArchive(ImageConfiguration imageConfiguration, BuildContext buildContext, String buildArchiveOnlyFlagOrBuildArchiveLocation)
```

Comment:
Can we make the last argument just a `archivePath` and if not-null, store the archive there ? The caller should then prepare that parameter based on the options given. That way this method becomes much more reusable and decoupled from its actual, Maven specific usage.